pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+	mutatingWebhookConfigurationGeneration int64
+	forceRollout                           bool
+}
+
+var _ reconcile.Reconciler = &staticResourceReconciler{}
+
+func (r *staticResourceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.WithField("controller", controllerName).Debugf("reconciling after watch event %#v", request)
+	if request.Name == "service-ca" && request.Namespace == "openshift-config-managed" {
+		log.WithField("controller", controllerName).Infof("%s configmap change detected", request.NamespacedName)
+		r.forceRollout = true
+	}
+	r.ReconcileResources()
+	return reconcile.Result{}, nil
+}
+
+func (r *staticResourceReconciler) ReconcileResources() {
+	applyResults := resourceapply.ApplyDirectly(
+		(&resourceapply.ClientHolder{}).WithKubernetes(r.clientset),
+		r.eventRecorder,
+		v410_00_assets.Asset,
+		staticFiles...,
+	)
+	for _, result := range applyResults {
+		if result.Error != nil {
+			log.WithField("controller", controllerName).WithError(result.Error).Errorf("error reconciling %s", result.Type)
+			continue
+		}
+		if !result.Changed {
+			continue
+		}
+		log.WithField("controller", controllerName).Infof("%s reconciled successfully", result.Type)
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/deployment.yaml"
+	requestedDeployment := resourceread.ReadDeploymentV1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/deployment.yaml"))
+	requestedDeployment.Spec.Template.Spec.Containers[0].Image = os.Getenv("AWS_POD_IDENTITY_WEBHOOK_IMAGE")
+	resultDeployment, modified, err := resourceapply.ApplyDeployment(r.clientset.AppsV1(), r.eventRecorder, requestedDeployment, r.deploymentGeneration, r.forceRollout)
+	r.forceRollout = false
+	r.deploymentGeneration = resultDeployment.Generation
+	if err != nil {
+		log.WithField("controller", controllerName).WithError(err).Error("error applying Deployment")
+	}
+	if modified {
+		log.WithField("controller", controllerName).Infof("Deployment reconciled successfully")
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml"
+	serviceCAConfigMap, err := r.clientset.CoreV1().ConfigMaps("openshift-config-managed").Get("service-ca", metav1.GetOptions{})
+	if err != nil {
+		log.WithField("controller", controllerName).WithError(err).Error("error reading service-ca configmap")
+	}
+	caBundle, ok := serviceCAConfigMap.Data["ca-bundle.crt"]
+	if !ok {
+		log.WithField("controller", controllerName).Error("ca-bundle.crt key not found service-ca configmap")
+	}
+	requestedMutatingWebhookConfiguration := ReadMutatingWebhookConfigurationV1Beta1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml"))
+	requestedMutatingWebhookConfiguration.Webhooks[0].ClientConfig.CABundle = []byte(caBundle)
+	resultMutatingWebhookConfiguration, modified, err := ApplyMutatingWebhookConfiguration(r.clientset.AdmissionregistrationV1beta1(), r.eventRecorder, requestedMutatingWebhookConfiguration, r.mutatingWebhookConfigurationGeneration)
```

Comment:
What is the purpose of storing and sending the mutatingWebhookConfigurationGeneration?

Reply:
That is the easiest way to detect if something external has changed the `MutatingWebhookConfiguration` as the apiserver will increment the `Generation`.  `ApplyDeployment` does the same thing (rather than try to DeepEquals the `spec` or something like that)

Reply:
I suppose my question comes from not seeing ApplyMutatingWebhookConfiguration() do anything with that final parameter that is passed to it.

Reply:
ah yes, very true.  in fact, i'm going to take that out completely in light @marun sharing that i can just add an annotation to the `MutatingWebhookConfiguration` and the `caBundle` will be managed for me.  thus external `Generation` changes would be expected.

Reply:
What was this annotation and feature that manages the caBundle for you? I am fighting similar problems over in Hive.

Reply:
n/m found it on slack.